### PR TITLE
components: introduce tirocks module

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?
Issue Number: Ref #13058 

What's Changed:
```commit-message
Only make it in the codebase, will not compile it. It will replace
engine_rocks once it's finished.
```

### Check List 

Tests 

- Manual test
   Uncomment the member list and `make dev`.

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/engine_tirocks/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+//! A new implementation of engine_traits using tirocks.
+//!
+//! When all features of engine_rocks are implemented in this module,
+//! engine_rocks will be removed and TiKV will switch to tirocks.
+
+extern crate tikv_alloc as _;
```
Comment (tabokie) [inline components/engine_tirocks/src/lib.rs:8]: Why is the `as _`?

**Review by tabokie — APPROVED** [review]

File: components/engine_tirocks/src/lib.rs (hunk 1 above)
Comment (BusyJay) [reply components/engine_tirocks/src/lib.rs:8]: It's only used for link purpose.

**Review by Connor1996 — APPROVED** [review]

**Pull request merged**